packages/glimmer-runtime/lib/environment.ts

```diff
@@ -116,9 +116,15 @@ export class Scope {
   }
 }
 
+interface ModifierPair {
```

Comment:
we should use two arrays like we do for components; in addition to the cost of allocating the object, it has some pretty bad issues with GC driven deopts